janitor/functions.py

```diff
@@ -4104,3 +4104,210 @@ def process_text(
     df[column] = getattr(df[column].str, string_function)(*args, **kwargs)
 
     return df
+
+
+@pf.register_dataframe_method
+def fill(
+    df: pd.DataFrame,
+    columns: Union[Tuple[str], List[str], str],
+    directions: Union[Tuple[str], List[str], str] = "down",
+    limit: Optional[int] = None,
+) -> pd.DataFrame:
```

Comment:
The `fill` function falls into the class of functions where we can trivially map column name to keyword if we want to express that the operation is applied over multiple columns, with potentially different directions on each column. This pattern is used in `find_replace`, and I think it can be done here too. (It seems clearer and clearer to me that where appropriate, this should be the pattern used.)

I think the most Python-idiom-respecting API might look something like this:

```python
def fill(
    df: pd.DataFrame,
    directions: Dict[Hashable, str],
    limit: Optional[int] = None,
) -> pd.DataFrame:
```

The use of a string that is split by commas is not something I would encourage. This is equivalent to passing in a mapping of x->y, and where mappings are used, I would advocate for dictionaries instead.